### ISSUE TITLE
Complex-valued vector expression dropping imaginary part

### ISSUE THREAD

**Issue by tikuma-lsuhsc** [cvxpy/cvxpy#2182]
**Describe the bug**

Using complex-valued constants with different precision drops the imaginary part from the resulting cvxpy expression.

**To Reproduce**

```python
import cvxpy as cp
import numpy as np

N = 100
M = 5

x = np.ones(N, "complex64")
A = np.ones((N, M), "complex128")

theta = cp.Variable(M, complex=True)
A @ theta - x
```

This produces a warning:

```
...\lib\site-packages\cvxpy\interface\numpy_interface\ndarray_interface.py:53: ComplexWarning: Casting complex values to real discards the imaginary part
  return result.astype(numpy.float64)
```

Using this expression to form a problem (obviously) yields an incorrect solution.

**Expected behavior**

Maintain the complex-valued expression at the highest precision of the constants.

**Version**
 - OS: Win10
 - CVXPY Version: v1.3.2

**Comment by SteveDiamond** [comment]
That's surprising we missed this case in our tests. I pushed a fix in #2183 with expanded test coverage.

**Comment by tikuma-lsuhsc** [comment]
👍 Nice. Thank you, and keep up the great work! Love this package.
